File: spring-java/src/main/java/org/distr/rcontacts/repository/sql/ContactRepository.java
package org.distr.rcontacts.repository.sql;

import org.distr.rcontacts.entities.ContactEntity;
import java.util.Optional;
import java.util.List;

public interface ContactRepository {

    public ContactEntity create(ContactEntity contact);
    public ContactEntity update(ContactEntity contact);
    public Optional<ContactEntity> findById(String id);
    public List<ContactEntity> findAllByUserId(String userId);
    public void delete(String id);

}

File: express/src/modules/contact/contact.repository.ts
import { Contact } from "./contact";

export default interface IContactRepository {
  create(userId: string, name: string, email: string, phone: string): Promise<Contact>;
  update(id: string, name: string, email: string, phone: string): Promise<Contact | null>;
  findById(id: string): Promise<Contact | null>;
  findAll(userId: string): Promise<Partial<Contact>[]>;
  delete(id: string): Promise<Contact> ;
}